src/main/java/org/hmhb/program/DefaultProgramService.java

```diff
@@ -274,10 +288,64 @@ public Program save(
             throw new ProgramPrimaryGoal1RequiredException();
         }
 
+        if (program.getPrimaryGoal1().length() > publicConfig.getProgramGoalMaxLength()) {
```

Comment:
This service is getting pretty dense.  A refactor is probably in order, but that can happen later -- at least for now, I have unit tests, so we can refactor and have confidence we're not totally screwing everything up.